hw/ip/pwm/doc/_index.md

```diff
@@ -0,0 +1,181 @@
+The blink-rate for each channel is independently defined relative to the common PWM pulse-frequency via a pair of 8-bit clock divider and clock pre-divider registers.
+The duty-cycle and phase of the blink rate can also be controlled, each with up to 8-bit resolution.
+Each channel also has a separate blink-enable register bit, allowing for multiple outputs to be configured to blink synchronously.
+
+# Theory of Operations
+
+## Hardware Interfaces
+
+{{< hwcfg "hw/ip/pwm/data/pwm.hjson" >}}
+
+## Design Details
+
+The IP consists of a single 16-bit phase counter and a number of PWM FSM instances, one per output.
+
+### PWM Phase Counter
+
+Each PWM pulse cycle is divided into 2<sup>16</sup> beats.
+During each beat, the 16-bit phase counter increments by one (modulo 65536).
+The beat period is defined by the {{< regref "CFG.CLK_DIV" >}} register:
+$$f_\textrm{beat}=\frac{f_\textrm{I/O clk}}{\textrm{CLK_DIV}+1}$$
+
+A PWM pulse cycle is completed each time the phase counter overflows to 0.
+The PWM drive frequency is therefore:
+$$f_\textrm{PWM}=\frac{f_\textrm{I/O clk}}{2^{16}(\textrm{CLK_DIV}+1)}$$
+
+The PWM phase counter is reset whenever {{< regref "CFG.CNTR_EN" >}} is disabled.
+
+The following figure illustrates the effect of the clock divider register.  Note that changes to {{< regref "CFG.CLK_DIV" >}} only take effect when {{< regref "CFG.CNTR_EN" >}} is disabled.
+
+{{< wavejson >}}
+{signal: [
+  {name: 'clk', wave: 'p..............|.........'},
+  {name: 'PWM_EN', wave: '0..1...........|01.......'},
+  {name: 'CLK_DIV', wave: '2....4.........|.2.......', data: '2 4 4'},
+  {name: 'clk_div_int', wave: '2................2.......', data: '2 4'},
+  {name: 'phase_ctr', wave: '3..2..2..2..2..|32....2..', data: '0 0 1 2 3 0 0 1'},
+  {name: 'clk_div_ctr', wave: '2...22222222222|2.2222222', data: '0 1 2 0 1 2 0 1 2 0 1 2 0 1 2 3 4 0 1 2'},
+  {name: 'clk_div_pulse', wave: '0....10.10.10.1|0....10..'}],
+  config:{skin:'narrow'}
+}
+{{< /wavejson >}}
+
+### PWM FSMs
+
+Each output is driven by an independent FSM.
+Each takes the phase-counter as a common input, as well as the following channel-specific register inputs:
+- a single PWM enable bit (e.g. {{< regref "PWM_EN.EN_0" >}})
+- a 16-bit PWM duty cycle (e.g. {{< regref "PWM_PARAM_0.PHASE_DELAY_0" >}})
+- a 16-bit PWM phase delay ({{< regref "PWM_PARAM_0.DUTY_CYCLE_0" >}})
+- a single blink enable bit ({{< regref "BLINK_EN.EN_0" >}})
+- a 16-bit blink rate divider (e.g. {{< regref "BLINK_PARAM_0.RATE_0" >}})
+- an 8-bit blink duty cycle (e.g. {{< regref "BLINK_PARAM_0.DELAY_0" >}})
+- an 8-bit blink phase adjust (e.g. {{< regref "BLINK_PARAM_0.DUTY_CYCLE_0" >}})
+
+In the operational description below, all of the examples discuss the operation of channel 0, using registers and fields ending with "_0".
+To operated other channels, simply choose the registers with the appropriate channel suffix.
```

Comment:
Stray "d": should be "To operate other channels..."

Reply:
Thanks.  Got it.